scripts/receipts.sh

```diff
@@ -104,4 +110,71 @@ function receipts::generate() {
   syft packages "${image}" --output cyclonedx-json --file "${output}"
 }
 
+# Generates syft receipts for each architecture for given oci archives
+function receipts::generate::multi::arch() {
+  local buildArchive runArchive registryPort registryPid localRegistry imageType archiveName imageReceipt
+
+  buildArchive="${1}"
+  runArchive="${2}"
+
+  registryPort=$(get::random::port)
+  localRegistry="127.0.0.1:$registryPort"
+
+  # Start a local in-memory registry so we can work with the oci archives
+  PORT=$registryPort crane registry serve --insecure > /dev/null 2>&1 &
+  registryPid=$!
+
+  # Wait for the registry to be available
+  until crane catalog $localRegistry > /dev/null 2>&1; do
+    sleep 1
+  done
+
+  # Push the oci archives to the local registry
+  jam publish-stack \
+    --build-ref "$localRegistry/build" \
+    --build-archive $buildArchive \
+    --run-ref "$localRegistry/run" \
+    --run-archive $runArchive
+  
+  # Ensure we can write to the BUILD_DIR
+  if [ $(stat -c %u build) = "0" ]; then
+    sudo chown -R "$(id -u):$(id -g)" "$BUILD_DIR"
```

Comment:
This is probably fine, but out of curiosity when do you run into issues where the `./build` directory isn't writable?

Reply:
The create-stack action [here](https://github.com/paketo-buildpacks/github-config/blob/main/actions/stack/create-stack/Dockerfile) runs inside an ubuntu:jammy container which runs as root, so the build directory and everything in it is owned by root as result.

Reply:
That makes sense, thanks. If you need this now, how was it working before?

Again, I'm not adverse to including this. I just don't (yet) understand why it's necessary now if it wasn't before.